cterasdk/core/cloudfs.py

```diff
@@ -157,28 +157,35 @@ def list_folders(self, include=None, list_filter=ListFilter.NonDeleted, user=Non
         param = builder.build()
         return query.iterator(self._portal, '/cloudDrives', param)
 
-    def find(self, name, owner, include):
+    def find(self, name, owner, include=None):
         """
-        Find a  Cloud Drive Folder
+        Find a Cloud Drive Folder
 
         :param str name: Name of the Cloud Drive Folder to find
         :param cterasdk.core.types.UserAccount owner: User account of the folder group owner
         :param list[str] include: List of metadata fields to include in the response
+
+        :returns: A Cloud Drive Folder
         """
+        include = union(include or [], CloudFS.default)
         builder = query.QueryParamBuilder().include(include)
-        query_filter = query.FilterBuilder('name').eq(name)
-        builder.addFilter(query_filter)
+        name_filter = query.FilterBuilder('name').eq(name)
+        owner_filter = query.FilterBuilder('owner', True).eq(owner.name)
+        builder.addFilter(name_filter).addFilter(owner_filter)
         param = builder.build()
 
-        iterator = query.iterator(self._portal, '/cloudDrives', param)
+        folders = list(query.iterator(self._portal, '/cloudDrives', param))
 
-        owner_ref = '/PortalUser/%s' % (owner.name) if owner.is_local else '/ADUser/%s/%s' % (owner.name, owner.directory)
-        for cloud_folder in iterator:
-            if cloud_folder.owner.endswith(owner_ref):
-                return cloud_folder
+        if len(folders) > 1:
```

Comment:
Why would you get more than one? And if you can get more than one, shouldn't you return all of them to the user? Won't this make the return of the method inconsistent?

Reply:
The backend doesn't allow you to filter based on domain + user. So if I lookup a directory by user, i may receive more than 1 result if the same user and folder name are used in different domains. I just updated the implementation to use a custom attribute I added long ago in query params called: `ownedBy`. Let's you pass a uid. So to guarantee uniqueness, I first lookup the user's uid and then proceed to query for the folder while passing the uid (the GUI does something similar)